inlang/source-code/sdk/src/loadProject.ts

```diff
@@ -46,6 +46,16 @@ const settingsCompiler = TypeCompiler.Compile(ProjectSettings)
 export async function loadProject(args: {
 	projectPath: string
 	nodeishFs: Repository["nodeishFs"]
+	/**
+	 * The app id is used to identify the app that is using the SDK.
+	 *
+	 * We use the app id to group events in telemetry to answer questions
+	 * like "Which apps causes these errors?" or "Which apps are used more than others?".
+	 *
+	 * @example
+	 * 	appId: "app.inlang.badge"
+	 */
+	appId?: string
```

Comment:
Shouldn't we simply make this mandatory?

Reply:
no. we don't want tests to have an appId

Reply:
Don't we have to do it sooner or later anyway because of the marketplace?

Reply:
No it's redundant. Telemetrty is not a hard requirement for the SDK to work. Forcing appids for the SDK would be wrong